[DAL] CSV Data Source specification (#199)
oneapi-src/oneAPI-spec: source/elements/oneDAL/source/data_management/csv_data_source.rst

.. highlight:: cpp
.. default-domain:: cpp

.. _csv-data-source:

---------------
CSV data source
---------------
Class ``csv::data_source`` is an API for accessing the data source represented as a :capterm:`csv file <CSV file>`.
CSV data source shall be used with :expr:`read` operation to extract data in text format from the given input file,
process it using provided parameters (such as delimiter and read options), transform it into numerical representation,
and store it as an in-memory :txtref:`dataset` of a chosen type.

> Supported type of in-memory dataset for :expr:`read` operation with CSV data source is :txtref:`table`.

Author: Done.

      static constexpr read_options default_read_options = read_options::default;

      data_source(const char *file_name,
                  char delimiter = default_delimiter,
                  read_options opts = default_read_options);

      data_source(const std::string &file_name,
                  char delimiter = default_delimiter,
                  read_options opts = default_read_options);

      std::string get_file_name() const;
      char get_delimiter() const;
      read_options get_read_options() const;
   };

.. namespace:: onedal::csv
.. class:: data_source

   .. function:: data_source(const char *file_name, char delimiter = default_delimiter, read_options opts = default_read_options)

      Creates a new instance of a CSV data source with the given :expr:`file_name`, :expr:`delimiter` and read options :expr:`opts` flag.

   .. function:: data_source(const std::string &file_name, char delimiter = default_delimiter, read_options opts = default_read_options);

      Creates a new instance of a CSV data source with the given :expr:`file_name`, :expr:`delimiter` and read options :expr:`opts` flag.

   .. member:: std::string file_name = ""

>       String which contains name of the file with the dataset to read.
> 
>       Getter
>          | ``std::string get_filename() const``
> 
>    .. member:: char delimiter = ','
> 
>       Symbol that represents delimiter between separate features in the input file.

Reviewer: ```suggestion
      A string that contains the name of the file with the dataset to read.

      Getter
         | ``std::string get_filename() const``

   .. member:: char delimiter = ','

      A character that represents the delimiter between separate features in the input file.
```
